Dnn.AdminExperience/Dnn.PersonaBar.Extensions/Services/ServerSettingsSmtpHostController.cs

```diff
@@ -1,190 +1,191 @@
+                }
+                else
+                {
+                    PortalController.UpdatePortalSetting(portalId, "SMTPServer", request.SmtpServer, false);
+                    PortalController.UpdatePortalSetting(portalId, "SMTPConnectionLimit", request.SmtpConnectionLimit, false);
+                    PortalController.UpdatePortalSetting(portalId, "SMTPMaxIdleTime", request.SmtpMaxIdleTime, false);
+                    PortalController.UpdatePortalSetting(portalId, "SMTPAuthentication",
+                        request.SmtpAuthentication.ToString(), false);
+                    PortalController.UpdatePortalSetting(portalId, "SMTPUsername", request.SmtpUsername, false);
+                    PortalController.UpdateEncryptedString(portalId, "SMTPPassword", request.SmtpPassword, Config.GetDecryptionkey());
+                    PortalController.UpdatePortalSetting(portalId, "SMTPEnableSSL", request.EnableSmtpSsl ? "Y" : "N", false);
+                }
+
+                DataCache.ClearCache();
+                return this.Request.CreateResponse(HttpStatusCode.OK, new { success = true });
+            }
+            catch (Exception exc)
+            {
+                Logger.Error(exc);
+                return this.Request.CreateErrorResponse(HttpStatusCode.InternalServerError, exc);
+            }
+        }
+
+        /// POST: api/Servers/SendTestEmail
+        /// <summary>
+        /// Tests SMTP settings.
+        /// </summary>
+        /// <param name="request"></param>
+        /// <returns></returns>
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public HttpResponseMessage SendTestEmail(SendTestEmailRequest request)
+        {
+            try
+            {
+                var smtpHostMode = request.SmtpServerMode == "h";
+                var mailFrom = smtpHostMode ? Host.HostEmail : this.PortalSettings.Email;
+                var mailTo = this.UserInfo.Email;
+
+                var errMessage = Mail.SendMail(mailFrom,
+                    mailTo,
+                    "",
+                    "",
+                    MailPriority.Normal,
+                    Localization.GetSystemMessage(this.PortalSettings, "EMAIL_SMTP_TEST_SUBJECT"),
+                    MailFormat.Text,
+                    Encoding.UTF8,
+                    "",
+                    "",
+                    smtpHostMode ? HostController.Instance.GetString("SMTPServer") : request.SmtpServer,
+                    smtpHostMode ? HostController.Instance.GetString("SMTPAuthentication") : request.SmtpAuthentication.ToString(),
+                    smtpHostMode ? HostController.Instance.GetString("SMTPUsername") : request.SmtpUsername,
+                    smtpHostMode ? HostController.Instance.GetEncryptedString("SMTPPassword", Config.GetDecryptionkey()) : request.SmtpPassword,
+                    smtpHostMode ? HostController.Instance.GetBoolean("SMTPEnableSSL", false) : request.EnableSmtpSsl);

```

Comment:
I would think that we always want the test email to use the visible settings, regardless of mode.